igvc_navigation/src/mapper/ros_mapper.cpp

```diff
@@ -0,0 +1,370 @@
+    ROS_ERROR("%s", ex.what());
+    return false;
+  }
+  catch (std::runtime_error &ex)
+  {
+    ROS_ERROR("runtime error");
+    ROS_ERROR("Runtime Exception at getOdomTransform: [%s]", ex.what());
+    return false;
+  }
+}
+
+template <>
+bool ROSMapper::getOdomTransform(const uint64 timestamp)
+{
+  ros::Time message_timestamp;
+  pcl_conversions::fromPCL(timestamp, message_timestamp);
+  return getOdomTransform(message_timestamp);
+}
+
+// Populates igvc_msgs::map message with information from sensor_msgs::Image and the timestamp from pcl_stamp
+/**
+ * Populates <code>igvc_msgs::map message</code> with information from <code>sensor_msgs::Image</code> and the
+ * timestamp from <code>pcl_stamp</code>
+ * @param[out] message message to be filled out
+ * @param[in] image image containing map data to be put into <code>message</code>
+ * @param[in] pcl_stamp time stamp from the pcl to be used for <code>message</code>
+ */
+void ROSMapper::setMessageMetadata(igvc_msgs::map &message, sensor_msgs::Image &image, uint64_t pcl_stamp)
+{
+  pcl_conversions::fromPCL(pcl_stamp, image.header.stamp);
+  pcl_conversions::fromPCL(pcl_stamp, message.header.stamp);
+  message.header.frame_id = "/odom";
+  message.image = image;
+  message.length = static_cast<unsigned int>(length_x_ / resolution_);
+  message.width = static_cast<unsigned int>(width_y_ / resolution_);
+  message.resolution = static_cast<float>(resolution_);
+  message.orientation = static_cast<float>(state_.yaw());
+  message.x = static_cast<unsigned int>(std::round(state_.x() / resolution_) + start_x_ / resolution_);
+  message.y = static_cast<unsigned int>(std::round(state_.y() / resolution_) + start_y_ / resolution_);
+  message.x_initial = static_cast<unsigned int>(start_x_ / resolution_);
+  message.y_initial = static_cast<unsigned int>(start_y_ / resolution_);
+}
+
+/**
+ * Checks if transform from base_footprint to msg.header.frame_id exists
+ * @param[in] msg
+ * @param[in] topic Topic to check for
+ */
+
+template <>
+bool ROSMapper::checkExistsStaticTransform(const std::string &frame_id, ros::Time message_timestamp,
+                                           const std::string &topic)
+{
+  if (transforms_.find(topic) == transforms_.end())
+  {
+    // Wait for transform between frame_id (ex. /scan/pointcloud) and base_footprint.
+    ROS_INFO_STREAM("Getting transform for " << topic << " from " << frame_id << " to /base_footprint \n");
+    if (tf_listener_->waitForTransform("/base_footprint", frame_id, message_timestamp, ros::Duration(3.0)))
+    {
+      tf::StampedTransform transform;
+      tf_listener_->lookupTransform("/base_footprint", frame_id, message_timestamp, transform);
+      transforms_.insert(std::pair<std::string, tf::StampedTransform>(topic, transform));
+      ROS_INFO_STREAM("Found static transform from " << frame_id << " to /base_footprint");
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Failed to find transform for " << topic << " from " << frame_id
```

Comment:
I don't think so. If you launch this node first before the node that actually publishes the state transform, I think it's reasonable for this node to wait until the transform becomes available instead of simply dying.